src/libraries/System.IO.FileSystem/tests/RandomAccess/FlushToDisk.cs

```diff
@@ -0,0 +1,103 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.IO.Pipes;
+using System.Security.Cryptography;
+using Microsoft.Win32.SafeHandles;
+using Xunit;
+
+namespace System.IO.Tests
+{
+    public partial class RandomAccess_FlushToDisk : RandomAccess_Base<long>
+    {
+        // Setting this to false disables the ThrowsArgumentOutOfRangeExceptionForNegativeFileOffset() test
+        // which is not applicable to FlushToDisk() since FlushToDisk() does not take a file offset parameter.
+        protected override bool UsesOffsets => false;
+
+        // Setting this to false disables the ThrowsNotSupportedExceptionForUnseekableFile() test which is not
+        // applicable to FlushToDisk() since FlushToDisk() DOES in fact support unseekable files.
+        protected override bool ThrowsForUnseekableFile => false;
+
+        protected override long MethodUnderTest(SafeFileHandle handle, byte[] bytes, long fileOffset)
+        {
+            // NOTE: tests for checking how FlushToDisk() deals with invalid arguments (e.g. a null handle)
+            // are implemented in the base class and work by calling this "MethodUnderTest" which we override
+            // here to call the FlushToDisk() method we want to test.
+            RandomAccess.FlushToDisk(handle);
+            return 0;
+        }
+
+        [Fact]
+        public void UpdatesFileLastWriteTime()
+        {
+            // Save test file path so we can refer to the same file throughout the test.
+            string testFilePath = GetTestFilePath();
+
+            // Generate random bytes to write to file. To ensure that flushing works correctly for a variety of
+            // sizes, we want to test with buffers that are smaller than a page (e.g. just 1 byte) and buffers
+            // that are several times larger than a page (e.g. up to 10 pages).
+            int byteCount = Random.Shared.Next(1, Environment.SystemPageSize * 10);
+            byte[] randomBytes = RandomNumberGenerator.GetBytes(byteCount);
+
+            // Create a new file and open it for writing.
+            using (SafeFileHandle handle = File.OpenHandle(testFilePath, FileMode.CreateNew, FileAccess.Write))
+            {
+                // Write random bytes to file.
+                RandomAccess.Write(handle, randomBytes, fileOffset: 0);
+
+                // Get the file time BEFORE flushing to disk.
+                DateTime fileTimeBeforeFlush = File.GetLastWriteTimeUtc(testFilePath);
```

Comment:
consider setting a last write time here that's a little older. otherwise you'll fail when the file system timestamp granularity isn't high enough. at an extreme, this test will fail on a FAT32 partition.

also, this test will fail if a daylight savings change or leap second happens during the test. obviously, that's unlikely, but several of our tests fail every time there's such an event as there's zillions of tests and they run all the time. to fix that, you can set a date that's a year ago, for example.

Reply:
Interesting points @danmoseley! In this test we are not explicitly setting the last write time so I can't control it in any way. When the file is flushed, **it is the operating system that updates the last write time** and the test is just checking that the time after flushing is greater than the time before flushing (or that they are at least equal, as explained in the comment near the `Assert` at the end).

The most important comment in the test says:

```csharp
 // Flush the file to disk. As a bare minimum test, this should work without throwing an
 // exception on all supported platforms. 
```

...i.e. we are mostly interested in the fact that flushing works without throwing an exception on all platforms. So I can remove all the date-related asserts to avoid the issues you raised. Please let me know if I should do this 👍